cdc server panic when a puller is being canceled

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

2. What did you expect to see?


3. What did you see instead?

```
panic: send on closed channel

goroutine 3593 [running]:
github.com/pingcap/ticdc/cdc/puller.(*EntrySorter).AddEntry(0xc00bdf8d80, 0xc167f486c0)
        github.com/pingcap/ticdc@/cdc/puller/entry_sorter.go:99 +0x1ac
github.com/pingcap/ticdc/cdc/puller.(*pullerImpl).SortedOutput.func1(0xc00bdf8d80, 0x22e96c0, 0xc00ad6c340, 0xc00b06e540, 0xc0002a52f0, 0x24, 0x0, 0x0)
        github.com/pingcap/ticdc@/cdc/puller/puller.go:124 +0x23d
created by github.com/pingcap/ticdc/cdc/puller.(*pullerImpl).SortedOutput
        github.com/pingcap/ticdc@/cdc/puller/puller.go:99 +0x1e4
{"level":"warn","ts":"2020-03-27T18:29:21.381+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-55c91b53-fe11-4488-9c66-8c16bcb3cc50/hw-dt-wms-warp1-tidb01:2379","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
{"level":"warn","ts":"2020-03-27T18:59:38.449+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-9938a4e1-aec9-49ed-86c9-ded3c4be4948/hw-dt-wms-warp1-tidb01:2379","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
{"level":"warn","ts":"2020-03-27T18:59:38.579+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-9938a4e1-aec9-49ed-86c9-ded3c4be4948/hw-dt-wms-warp1-tidb01:2379","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
```

4. Versions of the cluster



    - TiCDC version (execute `cdc version`):

$ ./cdc version
Release Version: v4.0.0-beta.2-21-geb39ca8-dev
Git Commit Hash: eb39ca8d2545c185cd2f4b209c97855e4544a1f3
Git Branch: master
UTC Build Time: 2020-03-27 10:20:29
Go Version: go version go1.13.3 linux/amd64

--- a/cdc/puller/entry_sorter.go
+++ b/cdc/puller/entry_sorter.go
@@ -62,7 +62,6 @@ func (es *EntrySorter) Run(ctx context.Context) {
 			case <-ctx.Done():
 				atomic.StoreInt32(&es.closed, 1)
 				close(es.output)
-				close(es.resolvedCh)
 				return
 			case resolvedTs := <-es.resolvedCh:
 				es.lock.Lock()

--- a/cdc/puller/puller.go
+++ b/cdc/puller/puller.go
@@ -98,6 +98,7 @@ func (p *pullerImpl) SortedOutput(ctx context.Context) <-chan *model.RawKVEntry 
 	sorter := NewEntrySorter()
 	go func() {
 		sorter.Run(ctx)
+		defer close(sorter.resolvedCh)
 		for {
 			be, err := p.chanBuffer.Get(ctx)
 			if err != nil {

--- a/cdc/sink/mqProducer/kafka.go
+++ b/cdc/sink/mqProducer/kafka.go
@@ -231,6 +231,11 @@ func newSaramaConfig(ctx context.Context, c KafkaConfig) (*sarama.Config, error)
 
 	config.Producer.Retry.Max = 10000
 	config.Producer.Retry.Backoff = 500 * time.Millisecond
+
+	config.Admin.Retry.Max = 100
+	config.Admin.Retry.Backoff = 500 * time.Millisecond
+	config.Admin.Timeout = 10 * time.Second
+
 	return config, err
 }
 

--- a/cdc/sink/mqProducer/kafka.go
+++ b/cdc/sink/mqProducer/kafka.go
@@ -232,7 +232,7 @@ func newSaramaConfig(ctx context.Context, c KafkaConfig) (*sarama.Config, error)
 	config.Producer.Retry.Max = 10000
 	config.Producer.Retry.Backoff = 500 * time.Millisecond
 
-	config.Admin.Retry.Max = 100
+	config.Admin.Retry.Max = 10000
 	config.Admin.Retry.Backoff = 500 * time.Millisecond
 	config.Admin.Timeout = 10 * time.Second
 

--- a/cdc/sink/mqProducer/kafka.go
+++ b/cdc/sink/mqProducer/kafka.go
@@ -174,11 +174,11 @@ func NewKafkaSaramaProducer(ctx context.Context, address string, topic string, c
 			partitionNum = 4
 			log.Warn("topic not found and partition number is not specified, using default partition number", zap.String("topic", topic), zap.Int32("partition_num", partitionNum))
 		}
+		log.Info("create a topic", zap.String("topic", topic), zap.Int32("partition_num", partitionNum), zap.Int16("replication_factor", config.ReplicationFactor))
 		err := admin.CreateTopic(topic, &sarama.TopicDetail{
 			NumPartitions:     partitionNum,
 			ReplicationFactor: config.ReplicationFactor,
 		}, false)
-		log.Info("create a topic", zap.String("topic", topic), zap.Int32("partition_num", partitionNum), zap.Int16("replication_factor", config.ReplicationFactor))
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
@@ -234,7 +234,7 @@ func newSaramaConfig(ctx context.Context, c KafkaConfig) (*sarama.Config, error)
 
 	config.Admin.Retry.Max = 10000
 	config.Admin.Retry.Backoff = 500 * time.Millisecond
-	config.Admin.Timeout = 10 * time.Second
+	config.Admin.Timeout = 20 * time.Second
 
 	return config, err
 }

--- a/kafka_consumer/main.go
+++ b/kafka_consumer/main.go
@@ -128,7 +128,7 @@ func waitTopicCreated(address []string, topic string, cfg *sarama.Config) error 
 		return errors.Trace(err)
 	}
 	defer admin.Close()
-	for i := 0; i <= 10; i++ {
+	for i := 0; i <= 30; i++ {
 		topics, err := admin.ListTopics()
 		if err != nil {
 			return errors.Trace(err)

